server/src/main/java/org/apache/druid/client/ImmutableDruidDataSource.java

```diff
@@ -109,6 +112,27 @@ public long getTotalSizeOfSegments()
     return totalSizeOfSegments;
   }
 
+  /**
+   * This method finds the fully overshadowed segments in this datasource
+   *
+   * @return set of overshadowed segments
+   */
+  public Set<SegmentId> getFullyOvershadowedSegments()
```

Comment:
The name of an expensive computation method should start with get-, which usually implies cheap, no-allocation method in Java. It can be compute-, determine-, or find-.